components/director/internal/destinationfetchersvc/client.go

```diff
@@ -0,0 +1,212 @@
+		AuthStyle: oauth2.AuthStyleInParams,
+	}
+	cert, err := tls.X509KeyPair([]byte(instanceConfig.Cert), []byte(instanceConfig.Key))
+	if err != nil {
+		return nil, errors.Errorf("failed to create destinations client x509 pair: %v", err)
+	}
+
+	transport := &http.Transport{
+		TLSClientConfig: &tls.Config{
+			Certificates: []tls.Certificate{cert},
+		},
+	}
+
+	mtlsClient := &http.Client{
+		Transport: transport,
+		Timeout:   apiConfig.Timeout,
+	}
+
+	ctx = context.WithValue(ctx, oauth2.HTTPClient, mtlsClient)
+
+	httpClient := cfg.Client(ctx)
+	httpClient.Timeout = apiConfig.Timeout
+
+	return &Client{
+		httpClient: httpClient,
+		apiConfig:  apiConfig,
+		apiURL:     instanceConfig.URL,
+	}, nil
+}
+
+// FetchTenantDestinationsPage returns a page of destinations
+func (c *Client) FetchTenantDestinationsPage(ctx context.Context, page string) (*DestinationResponse, error) {
+	url := c.apiURL + c.apiConfig.EndpointGetTenantDestinations
+	req, err := c.buildRequest(ctx, url, page)
+	if err != nil {
+		return nil, err
+	}
+
+	log.C(ctx).Infof("Getting destinations page: %s data from: %s\n", page, url)
+
+	res, err := c.sendRequestWithRetry(req)
+	if err != nil {
+		return nil, err
+	}
+	defer res.Body.Close()
+
+	if res.StatusCode != http.StatusOK {
+		return nil, errors.Errorf("received status code %d when trying to fetch destinations", res.StatusCode)
+	}
+
+	var destinations []model.DestinationInput
+	if err := json.NewDecoder(res.Body).Decode(&destinations); err != nil {
+		return nil, errors.Wrap(err, "failed to decode response body")
+	}
+
+	pageCount := res.Header.Get(c.apiConfig.PagingCountHeader)
+	if pageCount == "" {
+		return nil, errors.Errorf("failed to extract header '%s' from destinations response", c.apiConfig.PagingCountParam)
+	}
+
+	return &DestinationResponse{
+		destinations: destinations,
+		pageCount:    pageCount,
+	}, nil
+}
+
+func (c *Client) buildRequest(ctx context.Context, url string, page string) (*http.Request, error) {
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, url, nil)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to build request")
+	}
+
+	query := req.URL.Query()
+	query.Add(c.apiConfig.PagingCountParam, "true")
+	query.Add(c.apiConfig.PagingPageParam, page)
+	query.Add(c.apiConfig.PagingSizeParam, strconv.Itoa(c.apiConfig.PageSize))
+	req.URL.RawQuery = query.Encode()
+	return req, nil
+}
+
+// FetchDestinationSensitiveData returns sensitive data of a destination
+func (c *Client) FetchDestinationSensitiveData(ctx context.Context, destinationName string) ([]byte, error) {
+	url := fmt.Sprintf("%s%s/%s", c.apiURL, c.apiConfig.EndpointFindDestination, destinationName)
+	log.C(ctx).Infof("Getting destination data from: %s \n", url)
+	req, err := http.NewRequest(http.MethodGet, url, nil)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to build request")
+	}
+
+	res, err := c.sendRequestWithRetry(req)
+	if err != nil {
+		return nil, err
+	}
+	defer res.Body.Close()
```

Comment:
defer func() {
		err := res.Body.Close()
		if err != nil {
			log.D().Warnf("Unable to close response body. Cause: %v", err)
		}
	}()